qa/integration-tests/src/test/java/io/zeebe/broker/it/network/AdvertisedAddressTest.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.broker.it.network;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.zeebe.client.ZeebeClient;
+import io.zeebe.client.ZeebeClientBuilder;
+import io.zeebe.client.api.response.Topology;
+import io.zeebe.containers.ZeebeContainer;
+import io.zeebe.containers.ZeebeGatewayContainer;
+import io.zeebe.containers.ZeebePort;
+import io.zeebe.containers.ZeebeTopologyWaitStrategy;
+import io.zeebe.test.util.asserts.TopologyAssert;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.ToxiproxyContainer;
+
+/**
+ * Use ToxiProxy. Even though it does not support UDP, we can still use ToxiProxy because only
+ * "gossip" messages use UDP. SWIM has other messages to probe and sync that uses TCP. So the
+ * brokers can still find each other.
+ */
+public class AdvertisedAddressTest {
+  private static final String TOXIPROXY_NETWORK_ALIAS = "toxiproxy";
+  private static final String TOXIPROXY_IMAGE = "shopify/toxiproxy:2.1.0";
+  private static final int CLUSTER_SIZE = 3;
+  private static final int PARTITION_COUNT = 1;
+  private static final int REPLICATION_FACTOR = 3;
+
+  @Rule public final Network network = Network.newNetwork();
+
+  @Rule
+  public final ToxiproxyContainer toxiproxy =
+      new ToxiproxyContainer(TOXIPROXY_IMAGE)
+          .withNetwork(network)
+          .withNetworkAliases(TOXIPROXY_NETWORK_ALIAS);
+
+  private List<ZeebeContainer> containers;
```

Comment:
`ZeebeContainer` use the embedded gateway - did you mean to use `ZeebeBrokerContainer`? I guess here we don't want the gateway to be behind the proxy since it's currently broken, right?

Reply:
It does not really matter. But since the test have standalone gateway, it would be better to use ZeebeBrokerContainer.